Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/security/SecurityHelper.kt

```diff
@@ -1,55 +1,36 @@
 package de.rki.coronawarnapp.util.security
 
-import android.annotation.SuppressLint
 import android.content.SharedPreferences
-import android.util.Base64
 import androidx.annotation.VisibleForTesting
 import de.rki.coronawarnapp.exception.CwaSecurityException
 import de.rki.coronawarnapp.util.di.AppInjector
 import de.rki.coronawarnapp.util.di.ApplicationComponent
-import de.rki.coronawarnapp.util.preferences.clearAndNotify
 import de.rki.coronawarnapp.util.security.SecurityConstants.ENCRYPTED_SHARED_PREFERENCES_FILE
-import timber.log.Timber
 
 /**
  * Key Store and Password Access
  */
 object SecurityHelper {
 
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    internal val encryptedPreferencesProvider: (ApplicationComponent) -> SharedPreferences = {
+    internal val encryptedPreferencesProvider: (ApplicationComponent) -> SharedPreferences? = {
```

Comment:
Please make the access & retry mechanism part of `EncryptedPreferencesHelper`, I think the whole `object SecurityHelper` can then be deleted`.  I think `EncryptionErrorResetTool` can then also be almost deleted, `EncryptedPreferencesFactory` should then likely also be simplified and made part of `EncryptedPreferencesHelper`.

Reply:
I moved `SecurityHelper` to `EncryptedPreferencesHelper` and `EncryptionErrorResetTool` was updated as well